cloud_images/centos7/dcos_vol_setup.sh

```diff
@@ -53,7 +53,12 @@ function main {
     echo "Adding entry to fstab: $fstab"
     echo "$fstab" >> /etc/fstab
     echo -n "Mounting: $device to $mount_location"
-    mount "$mount_location"
+    until grep ^$device /etc/mtab > /dev/null; do sleep 1; echo -n .; mount "$mount_location" || :; done
+    if [ "$mount_location" = "/var/log" ]; then
+      echo "Recreating /var/log/journal"
+      mkdir /var/log/journal
+      systemd-tmpfiles --create --prefix /var/log/journal
+    fi
```

Comment:
Btw. something I just also ran into is https://www.freedesktop.org/software/systemd/man/systemd.exec.html#IgnoreSIGPIPE=
which I'm wondering why it doesn't apply in our case. I did a test on Fedora 25 and there the issue is non-existent, SIGPIPE is being ignored by default. On CentOS 7 however even explicitly setting IgnoreSIGPIPE=true doesn't do anything. Though looking at the `systemd.exec` manpage the option is documented in that version of systemd.